pnparser/pnparser.py

```diff
@@ -0,0 +1,191 @@
+from redbot.core import commands
+
+import rivercogutils as RiverCogUtils
+import re as Regex
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from river_mwclient.esports_client import EsportsClient
+    from discord.guild import Guild
+    from redbot.core.bot import Red
+
+
+CURRENT_VERSION: str = "0.5.0"
+            
+
+class PatchNotesParser(commands.Cog):
+    """Parses League of Legends patch notes to Wiki format"""
+
+    def __init__(self, bot: 'Red') -> None:
+        self.bot: 'Red' = bot
+        self.patch_notes: 'PatchNotes | None' = None
+        
+        guild: 'Guild | None' = bot.get_guild(529775376721903617)
+        if guild is not None:
+            for channel in guild.text_channels:
+                if channel.name == "feature-bug-disc":
+                    self.bug_fix_channel = channel
+                    break
+    
+    async def __auto_report(self, ctx: GuildContext, message: str, patch_notes: PatchNotes) -> None:
+        # TODO: remove embeded patch notes info
+        await Tunnel.message_forwarder(destination=self.bug_fix_channel,
+                                       content="Patch notes parser **auto report** from command "
+                                       f"`{ctx.message.system_content}` executed at "
+                                       f"*{ctx.guild.name} - #{ctx.channel.name}:*\n"
+                                       f"https://discord.com/channels/{ctx.guild.id}/"
+                                       f"{ctx.channel.id}/{ctx.message.id}\n"
+                                       f"```fix\n{message}\n```"
+                                       "\n**Additional information**\n"
+                                       f"Patch notes: {patch_notes.patch_url}\n```python\n"
+                                       f"Cog version: {CURRENT_VERSION}\n"
+                                       f"Data Dragon version: {Dragon.current_version}\n```")
+        await ctx.send("Something went wrong and I could not parse.\n"
+                       "An automatic error report was generated and "
+                       "someone will look into it.")
+
+    def __validate_patch(self, patch_version: str) -> bool:
+        if Regex.search(r'^\s*[1-9]{1,2}(\.|,|-)[1-9]{1,2}\s*$', patch_version):
+            self.patch_version = patch_version.strip().replace(',', '.').replace('-', '.')
+            return True
+        return False
+
+    @commands.group()
+    async def pnparser(self, ctx: GuildContext) -> None:
+        """A League of Legends patch notes parser"""
+        pass
+
+    @pnparser.command()
+    async def version(self, ctx: GuildContext) -> None:
+        """Show package information"""
+        await ctx.send(f"Current version is {CURRENT_VERSION}.")
+
+    @pnparser.command()
+    async def report(self, ctx: GuildContext, *message: str) -> None:
+        """Report parser issues"""
+        await Tunnel.message_forwarder(destination=self.bug_fix_channel,
```

Comment:
check out how to work with Config objects and make the bug_fix_channel configurable

you'll have do like `self.config.register_global(bug_fix_channel = None)` in the init

and then `bug_fix_channel = await self.config.bug_fix_channel()` and make sure it was actually set etc

Reply:
I've been looking at this and it seems pretty straight forward, but how do you set an identifier? Apparently it should be a hard coded random integer. should I just get a number from a random integer generator?

Reply:
Just need to know about that identifier though.